notification/search.go

```diff
@@ -217,35 +253,69 @@ func (s *Store) TimeSeries(ctx context.Context, opts TimeSeriesOpts) ([]TimeSeri
 	}
 	defer rows.Close()
 
-	counts := make(map[int]int)
+	buckets := make(map[bucketID]int) // bucketID{}: count
 	for rows.Next() {
-		var index, count int
-		err := rows.Scan(&index, &count)
-		if err != nil {
-			return nil, err
+		var bucket, count int
+		var segmentLabel sql.NullString
+
+		if opts.SegmentBy != "" {
+			err := rows.Scan(&bucket, &count, &segmentLabel)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			err := rows.Scan(&bucket, &count)
+			if err != nil {
+				return nil, err
+			}
+		}
+		label := segmentLabel.String
+		if label == "" {
+			label = "Message Counts"
```

Comment:
Should we handle this in the UI? If `label` is empty, there are no segments, which feels a little misleading.

Reply:
Since the query handles both segmentation and all data, if there are no segments then the total message count is returned for each time series.

Reply:
Right, with no segmentation (i.e., there is a single group for all datapoints)

Reply:
I thought a default value for the label would be a nice ux if it's returning all counts, API-wise as well. Would there be a better way to represent that here?